pkg/services/sqlstore/migrations/ualert/migration_test.go

```diff
@@ -0,0 +1,602 @@
+package ualert_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"xorm.io/xorm"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrations"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrations/ualert"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/services/sqlstore/sqlutil"
+	"github.com/grafana/grafana/pkg/setting"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_AddDashAlertMigration(t *testing.T) {
```

Comment:
Conventional style for test names is `TestAddDashAlertMigration`. I see in some of these files that we have things like `Test_validateAlertmanagerConfig`, which tests the unexported function `validateAlertManagerConfig` for tests that are either about exported functions, or about general behavior, I'd recommend leaving out the `_`.